src/azul/service/source_service.py

```diff
@@ -94,9 +97,14 @@ def _dynamodb(self):
         return aws.dynamodb
 
     def _get(self, key: str) -> list[AnyJSON]:
+        log.debug('_request(GET, DynamoDB, key=%r) …', key)
+        start = time()
         response = self._dynamodb.get_item(TableName=self.table_name,
                                            Key={self.key_attribute: {'S': key}},
                                            ProjectionExpression=','.join([self.value_attribute, self.ttl_attribute]))
+        log.debug('_request(…) -> %.3fs %r',
+                  time() - start,
+                  trunc_ellipses(json.dumps(response), 128))
```

Comment:
This looks like a good opportunity to use `azul.json.json_head` to avoid serializing the entirety of a potentially large object:

```suggestion
                  trunc_ellipses(json_head(response, 128 + 1), 128))
```

The `+1` to ensure that `trunc_ellipses` stills works as intended if the string is longer than 128 characters.